src/openloco/ui.cpp

```diff
@@ -760,23 +767,116 @@ namespace openloco::ui
         SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT, title.c_str(), message.c_str(), window);
     }
 
-    bool updateSDLResolution()
+    static config::resolution_t getDisplayResolutionByMode(config::screen_mode mode)
     {
-        SDL_DisplayMode mode;
-        SDL_GetWindowDisplayMode(window, &mode);
+        config::resolution_t defaultResolution = { 640, 480 };
```

Comment:
For fullscreen mode, the fallback should be the window's minimal resolution, which may not be 640x480.

Windows should probably fall back to something bigger; I believe 800x600 is currently the default.